lib/galaxy/tool_util/parser/xml.py

```diff
@@ -502,6 +504,13 @@ def parse_help(self):
     def macro_paths(self):
         return self._macro_paths
 
+    def paths_and_modtimes(self):
+        if not self._paths_and_modtimes:
```

Comment:
Is it safe to store this in the `XmlToolSource` object or is it better to regenerate the return value at each call?

Reply:
We're not re-using it right now, so yes, we can do that.